start

```diff
@@ -22,484 +22,544 @@ export PGPORT=5432
-  fi
+	if [ ! -z "$CURRENT_POSTGRES_PID" ]; then
+		return 0
+	fi
 
-  sudo -u postgres $PGBIN/postgres -D $PGDATA -c config_file=$PGHOME/etc/postgresql.conf &> /dev/null &
-  CURRENT_POSTGRES_PID=$!
+	sudo -u postgres $PGBIN/postgres -D $PGDATA -c config_file=$PGHOME/etc/postgresql.conf &> /dev/null &
+	CURRENT_POSTGRES_PID=$!
 
-  while ! sudo -u postgres psql -c 'select 1' &> /dev/null ; do
-    echo "Waiting for postgres to be available..."
-    sleep 1
-  done
+	while ! sudo -u postgres psql -c 'select 1' &> /dev/null ; do
+		print_output "Waiting for postgres to be available..."
+		sleep 1
+	done
 
-  echo "postgres: up"
+	print_output "postgres: up"
 }
 
 function stop_postgres() {
-  if [ -z "$CURRENT_POSTGRES_PID" ]; then
-    return 0
-  fi
-
-  killall postgres
-  # wait for postgres to die
-  while kill -0 "$CURRENT_POSTGRES_PID" &> /dev/null; do
-    sleep 0.5
-  done
-  echo "postgres: down"
+	if [ -z "$CURRENT_POSTGRES_PID" ]; then
+		return 0
+	fi
+
+	killall postgres
+	# wait for postgres to die
+	while kill -0 "$CURRENT_POSTGRES_PID" &> /dev/null; do
+		sleep 0.5
+	done
+	print_output "postgres: down"
 }
 
 pushd () {
-    command pushd "$@" > /dev/null
+	command pushd "$@" > /dev/null
 }
 
 popd () {
-    command popd "$@" > /dev/null
+	command popd "$@" > /dev/null
+}
+
+function soroban_rpc_status () {
+	horizon_status
+	print_colored_screen_output "waiting for soroban rpc to report ready state..." "\033[34m" 
+	COUNTER=1
+	while ! $(curl --silent --location --request POST 'http://localhost:8000/soroban/rpc' \
+		--header 'Content-Type: application/json' \
+		--data-raw '{
+				"jsonrpc": "2.0",
+				"id": 10235,
+				"method": "getHealth"
+				
+		}' | jq --exit-status '.result.status == "healthy"' 2>/dev/null | grep -o true || echo false ); 
+	do
+		if [ $COUNTER -gt 12 ]; then 
+			print_colored_screen_output "waiting for soroban rpc to report ready state..." "\033[34m" 
+			COUNTER=1
+		else 
+			COUNTER=$[$COUNTER +1]    
+		fi 
+		sleep 5
+	done 
+	print_colored_screen_output "soroban rpc reported ready status, the service can be used by tools/cli now ..." "\033[32m" 
+}
+
+function horizon_status () {
+	COUNTER=1
+	print_colored_screen_output "waiting for horizon ingestion to be caught up to network current ledger..." "\033[34m" 
+	while ! $(curl --silent --location --request GET 'http://localhost:8000' | \
+		jq --exit-status '.core_latest_ledger > 5 and .history_latest_ledger > 5 and (.core_latest_ledger - .history_latest_ledger) < 5' 2>/dev/null | \
+		grep -o true || echo false ); 
+	do
+		if [ $COUNTER -gt 12 ]; then 
+			print_colored_screen_output "waiting for horizon ingestion to be caught up to network current ledger..." "\033[34m" 
+			COUNTER=1
+		else 
+			COUNTER=$[$COUNTER +1] 
+		fi 
+		sleep 5
+	done 
+	print_colored_screen_output "horizon ingestion is caught up to network current ledger ..." "\033[32m" 
+}
+
+function print_colored_screen_output() {
+	echo -e "$2$1\033[0m" | tee -a $QUICKSTART_LOG
```

Comment:
This PR seems to be evolving a bit, and I noticed it now has colors too. I'm all for colors that assist in extremely noisy environments, however the quickstart image's output is very small. I tried it out, and as a user the colors don't feel like they are serving a useful purpose. Take this for example:

<img width="1357" alt="Screenshot 2023-01-20 at 4 26 09 PM" src="https://user-images.githubusercontent.com/351529/213829542-3b3723f2-276f-4623-9693-e7129baf7b8c.png">

The red/orange line implies to me that something is wrong, but the message says waiting, so if I imagine myself as someone who doesn't know how it works internally, it's somewhat confusing.

The green line implies it is good, but again, on the backdrop of very little noise the colors don't add much value.

I'd reserve colors for something like #405, where there will be a ton of very noisy logs.

Reply:
yea, I haven't used console colors much, that orange was supposed to be blue, at least that was the escape code for blue. I removed coloring.